aten/src/ATen/native/native_functions.yaml

```diff
@@ -2008,6 +2008,9 @@
 - func: type_as(Tensor self, Tensor other) -> Tensor
   variants: method
 
+- func: has_same_tensorimpl_type_as(Tensor self, Tensor other) -> bool
```

Comment:
can we:
1) make this a function.
2) start this with an underscore (it's not really part of the public API).